tests/cpp/helpers.cc

```diff
@@ -142,4 +143,38 @@ std::shared_ptr<xgboost::DMatrix>* CreateDMatrix(int rows, int columns,
   return static_cast<std::shared_ptr<xgboost::DMatrix> *>(handle);
 }
 
+std::unique_ptr<DMatrix> CreateSparsePageDMatrix() {
+  // Create sufficiently large data to make two row pages
+  dmlc::TemporaryDirectory tempdir;
+  const std::string tmp_file = tempdir.path + "/big.libsvm";
+  CreateBigTestData(tmp_file, 12);
+  DMatrix * dmat = DMatrix::Load(
+      tmp_file + "#" + tmp_file + ".cache", true, false, "auto", 64UL);
```

Comment:
Let's wrap pointer in the smart pointer immediately, for the sake of exception safety.
```cpp
std::unique_ptr<DMatrix> dmat = std::unique_ptr<DMatrix>(DMatrix::Load(
      tmp_file + "#" + tmp_file + ".cache", true, false, "auto", 64UL));
```

Reply:
Done.